pyramid_deferred_sqla/__init__.py

```diff
@@ -215,6 +220,43 @@ def _create_engine(config, db_url_key="sqlalchemy.url", **kw):
     Base.metadata.bind = engine
     config.registry.settings["sqlalchemy.engine"] = engine
 
+    check_db_migrated(config)
+
+
+def check_db_migrated(config: Configurator) -> None:
+    """Check if DB is migrated to Alembic head.
+
+    Do it by comparing the Alembic revision in DB to the Alembic revision on
+    the filesystem.
+    """
+
+    # skip if SKIP_CHECK_DB_MIGRATED is set
+    if config.registry.settings.get("SKIP_CHECK_DB_MIGRATED"):
+        return
+
+    # skip if we are bootstrapping from alembic env, meaning when running
+    # an alembic command in the terminal
+    if getattr(alembic.context, "config", None):
+        return
+
+    # get latest migration file
+    alembic_config = alembic.config.Config("etc/alembic.ini")
+
+    script = ScriptDirectory.from_config(alembic_config)
+    head = EnvironmentContext(alembic_config, script).get_head_revision()
+
+    # get latest version from db
+    with config.registry.settings["sqlalchemy.engine"].connect() as conn:
+        curr = MigrationContext.configure(conn).get_current_revision()
+
+    if curr != head:
+        sys.exit(
```

Comment:
Functional languages are "testable" because they have (or expected to, at least) no "side effects".
"exit" is a pretty big side effect IMO.
returning "passed/failed" from the function and taking action (like `sys.exit`) in the caller might make this better code.